### DISCUSSION TITLE
Idea Discussion - First Critical Task

### DISCUSSION THREAD

**Discussion by evnd04** [uprm-inso4101-2023-2024-s2/semester-project-rumaradoptionapp#1] [Ideas]
By the end of the first week, once all the students have joined a team, the goal is for every
member of the team to contribute at least one idea for a task/feature/action item. This idea
should be created as a discussion post on the Github Repo to allow members to easily
comment on it. This idea/feature should eventually be decently fleshed out; it should lay out the
user need it aims to solve, the requirements for this need to be satisfied, and the expected
implementation requirements (like how many developers you think you will need). But even
though eventually it should be fleshed out, initially it can be just a thought/question/…

**Comment by rukitoyordan** [comment]
Good evening everyone! I'd like to contribute on some ideas here to get the ball rolling on some discussion. Let's think briefly on Main/Primary Features. 📍

### **Ideas for Main Features:**

1. Sign Up/Login Feature ✏️
Any self-sufficient application or website should have the capabilities to store user information and recall it when needed. 

2. Home Screen/Site 🏠
In the early stages, we should focus on making things work, rather than making them look pretty right off the bat. If it can be done simultaneously, that's great! Otherwise, the app/website's performance is the priority in the early stages. (Can't wait to see your creativity and thought processes throughout the semester)

3. Database for Storing Information 💻

### Ideas for Tasks 💡
Some groups can be divided for these main features and have more discussion on tasks. Remember that tasks are the exoskeleton for development as well as a gauge for how your efforts may be graded. Take seriously your task ideas as well as set realistic goals.

For example:
**Sign Up/Login Feature**
(Main) 👁️
- [ ] Create a Text Box that is centralized 
- [ ] Add labels and input types for username 
- [ ] Add labels and input types for password
- [ ] Create a link for Terms and Conditions
- [ ] Add a Submit and Cancel button

(Optional) ✨
- [ ] Style your Sign Up/Login Box
- [ ] Change Button Opacity or create an Animation when hovering
- [ ] Add a sound effect to the buttons
- [ ] Change text to red if it doesn't match passwords
- [ ] Change text to green if username is available

These are just some basic ideas (I have more but we want to hear from everyone too!) for Main/Primary Features and task organization. These first steps for features are key to a successful project. There can also of course be room for anyone who wishes to input some creative ideas regarding UI, Buttons, code efficiency, prioritization of tasks or features, you name it! One step at a time, and we can build a masterpiece. 🙌🏽

Please reply with any further ideas or comments on how can this ideas be expanded upon as tasks for our first Milestone ~

**Comment by evnd04** [comment]
The other features to be included in the first milestone should be profiles, specifically user and pet profiles

User and Pet Profiles Feature:
  ### Main Tasks:
      
  
  - [ ] *Create User Profile*: 
  

    -  Design and implement a user profile section.
    -  Include fields like name, email, and profile picture.

  
  - [ ]  *Create Pet Profile*:
  

    -  Design and implement a pet profile section.
    -  Include fields like name, age, breed, and photos.

  
  - [ ]  *Connect User and Pet Profiles*:
 

    - Establish a relationship between user and pet profiles.
    - Ensure proper data association and retrieval.

  
  - [ ] *Add Edit Profile Functionality*:

    - Implement the ability for users to edit their profiles.
    - Include validations for profile updates.

  
  - [ ] *Implement Profile Display*:

    - Display user and pet profiles on the app/site.
    - Ensure a clean and organized layout.

  
  ### Optional Tasks:
  
  - [ ] *Style User and Pet Profiles*:

    -  Apply advanced CSS for improved aesthetics.
    -  Consider responsive design for different devices.

  
  - [ ]  *Implement Profile Picture Upload*:
 
    - Allow users to upload profile pictures.
    - Implement server-side logic for image storage.

  
  - [ ]  *Add Animation on Profile Display*:   

    - Introduce animations for a visually appealing display.
    - Consider transitions between user and pet profiles.

  
  - [ ] *Include Pet Adoption History*:

    -  Display a history of adopted pets in the user profile.
    -  Enhance user engagement with meaningful content.
